src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -1132,6 +1133,29 @@ extern "C" int32_t SystemNative_INotifyRemoveWatch(intptr_t fd, int32_t wd)
 #endif
 }
 
+extern "C" int32_t SystemNative_GetPeerID(intptr_t socket, uid_t* euid)
+{
+    int fd = ToFileDescriptor(socket);
+#ifdef SO_PEERCRED
+    struct ucred creds;
+    socklen_t len = sizeof(creds);
+    if (getsockopt(fd, SOL_SOCKET, SO_PEERCRED, &creds, &len) == 0)
+    {
+        *euid = creds.uid;
+        return 0;
+    }
+    return -1;
+#elif HAVE_GETPEEREID
+    uid_t egid;
+    return getpeereid(fd, euid, &egid);
+#else
+    (void)fd;
+    (void)*euid;
+    errno = ENOTSUP;
+    return -1;
+#endif
+}
```

Comment:
This is almost entirely duplicating SystemNative_GetPeerUserName.  Could we rewrite that function in terms of this one to avoid the duplication? e.g.
```C++
extern "C" char* SystemNative_GetPeerUserName(intptr_t socket)
{
    uid_t euid;
    return SystemNative_GetPeerID(socket, &euid) == 0 ?
        GetNameFromUid(euid) :
        nullptr;
}
```